jni/src/faiss_wrapper.cpp

```diff
@@ -223,6 +226,76 @@ void knn_jni::faiss_wrapper::CreateIndexFromTemplate(knn_jni::JNIUtilInterface *
     faiss::write_index(&idMap, indexPathCpp.c_str());
 }
 
+void knn_jni::faiss_wrapper::CreateBinaryIndexFromTemplate(knn_jni::JNIUtilInterface * jniUtil, JNIEnv * env, jintArray idsJ,
```

Comment:
Can we refactor this method into 3 pieces to make more readable.

1. Validation 
2. AddPoints to Graph and WritertoDisk,

Reply:
Refactor isn't in scope of this PR.

Reply:
I would prefer Rule of 30 , if functions length is going over 30 lines , we should sperate logical

Reply:
there will have a separate PR for the refactoring, created issue https://github.com/opensearch-project/k-NN/issues/1846